lib/Parser.js

```diff
@@ -1178,7 +1178,10 @@ class Parser extends Tapable {
 
 	walkExportDefaultDeclaration(statement) {
 		this.hooks.export.call(statement);
-		if (statement.declaration.id) {
+		if (
+			statement.declaration.id &&
+			statement.declaration.type !== "FunctionExpression"
+		) {
```

Comment:
It's possible that this was an attempt to workaround an old acorn bug where it'd incorrectly parse `export default function () {}` as a function expression.

Reply:
In acorn 4, `export default function() {` was parsed as a `FunctionExpression`. 
Since acorn 5 it is a `FunctionDeclaration`.